webforces/server/mongodbworker.py

```diff
@@ -312,6 +312,29 @@ def getAllAlgs(self) -> Tuple[DBStatus, List[Algorithm]]:
         logger.debug("Algorithms was found")
         return (DBStatus.s_ok, algs)
 
+    def getAllAlgsWithFlags(self, user_id) -> Tuple[DBStatus, List[Algorithm], List[bool]]:
+        try:
+            st, user = self.getUserByID(user_id)
+            if st == DBStatus.s_data_issue:
+                logger.error("This user does not exist")
+                return (DBStatus.s_data_issue, [Algorithm(ERROR_ID)], [ERROR_ID])
+            algs_collection = self.db["algs"]
+            algs_tmp = list(algs_collection.find({}))
+            algs = []
+            flags = []
```

Comment:
Could you provide more explicit description what these flags mean? Maybe it's better to rename this variable and function?
It can be something like `is_owned_by_user` or whatever.

Reply:
Never mind :)

This function is needed for a specific occasion. Sergey knows how to use it.

Reply:
Well, use case for this function is pretty common. I'm pretty sure it's gonna be needed on desktop app too. I think it's better to give it descriptive name since this function is exposed by DBWorker interface.

Reply:
Done.